examples/platforms/utils/link_metrics.cpp

```diff
@@ -0,0 +1,285 @@
+        {
+            aData[bytes++] = static_cast<uint8_t>((GetAverageRss() + 130) * 255 /
+                                                  130); // Linear scale RSSI from [-130, 0] to [0, 255]
+        }
+
+    exit:
+        return bytes;
+    }
+
+    /**
+     * This method gets the metrics configured for the Enhanced-ACK Based Probing.
+     *
+     * @returns  The metrics configured.
+     *
+     */
+    otLinkMetrics GetLinkMetrics(void) const { return mLinkMetrics; }
+
+private:
+    LinkMetricsDataInfo *mNext;
+
+    otLinkMetrics mLinkMetrics;
+
+    otShortAddress mShortAddress;
+    otExtAddress   mExtAddress;
+
+    RssAverager mRssAverager;
+    LqiAverager mLqiAverager;
+
+    bool Matches(const otShortAddress &aShortAddress) const { return mShortAddress == aShortAddress; };
+
+    bool Matches(const otExtAddress &aExtAddress) const
+    {
+        return memcmp(&mExtAddress, &aExtAddress, sizeof(otExtAddress)) == 0;
+    };
+};
+
+enum
+{
+    kMaxEnhAckProbingInitiator = 10,
+};
+
+typedef Pool<LinkMetricsDataInfo, kMaxEnhAckProbingInitiator> LinkMetricsDataInfoPool;
+
+typedef LinkedList<LinkMetricsDataInfo> LinkMetricsDataInfoList;
+
+static LinkMetricsDataInfoPool &GetLinkMetricsDataInfoPool(void)
+{
+    static LinkMetricsDataInfoPool sDataInfoPool;
+    return sDataInfoPool;
+}
+
+static LinkMetricsDataInfoList &GetLinkMetricsDataInfoActiveList(void)
+{
+    static LinkMetricsDataInfoList sDataInfoActiveList;
+    return sDataInfoActiveList;
+}
+
+static inline bool IsLinkMetricsClear(const otLinkMetrics aLinkMetrics)
+{
+    return !aLinkMetrics.mPduCount && !aLinkMetrics.mLqi && !aLinkMetrics.mLinkMargin && !aLinkMetrics.mRssi;
+}
+
+void otLinkMetricsInit(const int8_t aNoiseFloor)
+{
+    sNoiseFloor = aNoiseFloor;
+}
+
+otError otLinkMetricsConfigureEnhAckProbing(const otShortAddress aShortAddress,
+                                            const otExtAddress * aExtAddress,
+                                            otLinkMetrics        aLinkMetrics)
+{
+    otError              error    = OT_ERROR_NONE;
+    LinkMetricsDataInfo *dataInfo = nullptr;
+
+    VerifyOrExit(aExtAddress != nullptr, error = OT_ERROR_INVALID_ARGS);
+
+    if (IsLinkMetricsClear(aLinkMetrics)) ///< Remove the entry
+    {
+        dataInfo = GetLinkMetricsDataInfoActiveList().RemoveMatching(aShortAddress);
+        VerifyOrExit(dataInfo != nullptr, error = OT_ERROR_NOT_FOUND);
+        GetLinkMetricsDataInfoPool().Free(*dataInfo);
+    }
+    else
+    {
+        dataInfo = GetLinkMetricsDataInfoActiveList().FindMatching(aShortAddress);
+
+        if (dataInfo == nullptr)
+        {
+            dataInfo = GetLinkMetricsDataInfoPool().Allocate();
+            VerifyOrExit(dataInfo != nullptr, error = OT_ERROR_NO_BUFS);
+            dataInfo->Clear();
+            GetLinkMetricsDataInfoActiveList().Push(*dataInfo);
+        }
+
+        dataInfo->Set(aLinkMetrics, aShortAddress, *aExtAddress);
```

Comment:
Are we overwriting a previous configuration by design? Maybe a comment about that here would be useful.